### ISSUE TITLE
In opacus.data_loader, batch_sampler option is mutually exclusive with drop_last

### ISSUE THREAD

**Issue by Varal7** [pytorch/opacus#352]
## 🐛 Bug

In opacus.data_loader, batch_sampler option is mutually exclusive with drop_last

## To Reproduce

```python
import torch
from torch.utils.data import DataLoader, TensorDataset
from opacus.data_loader import DPDataLoader
x, y = torch.randn(63, 5), torch.randint(0, 2, (63,))
dataset = TensorDataset(x,y)
data_loader = DataLoader(dataset, batch_size=4, drop_last=True)
dp_data_loader = DPDataLoader.from_data_loader(data_loader)
```

```
ValueError: batch_sampler option is mutually exclusive with batch_size, shuffle, sampler, and drop_last
```

## Expected behavior

Opacus should issue a warning that `drop_last` is not compatible with `DPDataLoader` and not propagate it to `DataLoader.__init__()`.

**Comment by romovpa** [comment]
@Varal7 thanks for reporting!
Fair enough. I agree we should ignore `drop_last` when converting DataLoader. 
I'm not sure if it's necessary to issue a warning though. The transformation is explicit, I think it's enough to make sure it is clear from the docstring.

**Comment by romovpa** [comment]
Fixed in #357
